### PR TITLE
Added supported v1 CLI parameters to v2

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: component/attr_cache/attr_cache.go [hunk 1]
```diff
@@ -131,7 +134,11 @@ func (ac *AttrCache) Configure(_ bool) error {
 		ac.cacheTimeout = defaultAttrCacheTimeout
 	}
 
-	ac.cacheOnList = !conf.NoCacheOnList
```
Comment (vibhansa-msft) [inline component/attr_cache/attr_cache.go:134]: We have not removed this NoCacheOnList flag here

**Review by vibhansa-msft — APPROVED** [review]

File: cmd/mount.go [hunk 2]
```diff
@@ -85,6 +86,11 @@ type mountOptions struct {
 	ProfilerPort      int            `config:"profiler-port"`
 	ProfilerIP        string         `config:"profiler-ip"`
 	MonitorOpt        monitorOptions `config:"health-monitor"`
+
+	// v1 support
+	Streaming      bool     `config:"streaming"`
+	AttrCache      bool     `config:"use-attr-cache"`
+	LibfuseOptions []string `config:"libfuse-options"`
```
Comment (souravgupta-msft) [inline cmd/mount.go:93]: We should add these in the baseConfig.yaml

File: cmd/mount.go [hunk 3]
```diff
@@ -283,6 +350,16 @@ var mountCmd = &cobra.Command{
 			os.Exit(1)
 		}
 
+		if config.IsSet("invalidate-on-sync") {
+			log.Warn("unsupported v1 CLI parameter: invalidate-on-sync is always true in blobfuse2.")
+		}
+		if config.IsSet("pre-mount-validate") {
+			log.Warn("unsupported v1 CLI parameter: pre-mount-validate is always true in blobfuse2.")
+		}
+		if config.IsSet("basic-remount-check") {
```
Comment (souravgupta-msft) [inline cmd/mount.go:359]: Should add these as well in baseConfig.yaml

File: component/attr_cache/attr_cache.go [hunk 4]
```diff
@@ -66,6 +66,9 @@ type AttrCacheOptions struct {
 	Timeout       uint32 `config:"timeout-sec" yaml:"timeout-sec,omitempty"`
 	NoCacheOnList bool   `config:"no-cache-on-list" yaml:"no-cache-on-list,omitempty"`
 	NoSymlinks    bool   `config:"no-symlinks" yaml:"no-symlinks,omitempty"`
+
+	// support v1
+	CacheOnList bool `config:"cache-on-list"`
```
Comment (souravgupta-msft) [inline component/attr_cache/attr_cache.go:71]: Same as previous

File: component/azstorage/config.go [hunk 5]
```diff
@@ -166,12 +167,18 @@ type AzStorageOptions struct {
 	AuthResourceString      string `config:"auth-resource" yaml:"auth-resource,omitempty"`
 	UpdateMD5               bool   `config:"update-md5" yaml:"update-md5"`
 	ValidateMD5             bool   `config:"validate-md5" yaml:"validate-md5"`
+
+	// v1 support
+	UseAdls        bool   `config:"use-adls"`
+	UseHTTPS       bool   `config:"use-https"`
+	SetContentType bool   `config:"set-content-type"`
+	CaCertFile     string `config:"ca-cert-file"`
```
Comment (souravgupta-msft) [inline component/azstorage/config.go:175]: Same as previous

File: component/file_cache/file_cache.go [hunk 6]
```diff
@@ -97,6 +97,10 @@ type FileCacheOptions struct {
 
 	EnablePolicyTrace bool `config:"policy-trace" yaml:"policy-trace,omitempty"`
 	OffloadIO         bool `config:"offload-io" yaml:"offload-io,omitempty"`
+
+	// v1 support
+	V1Timeout     uint32 `config:"file-cache-timeout-in-seconds"`
+	EmptyDirCheck bool   `config:"empty-dir-check"`
```
Comment (souravgupta-msft) [inline component/file_cache/file_cache.go:103]: Same as previous

File: component/stream/stream.go [hunk 7]
```diff
@@ -61,6 +61,10 @@ type StreamOptions struct {
 	BufferSizePerFile uint64 `config:"handle-buffer-size-mb" yaml:"handle-buffer-size-mb,omitempty"`
 	HandleLimit       uint64 `config:"handle-limit" yaml:"handle-limit,omitempty"`
 	readOnly          bool   `config:"read-only"`
+
+	// v1 support
+	StreamCacheMb    uint64 `config:"stream-cache-mb"`
+	MaxBlocksPerFile uint64 `config:"max-blocks-per-file"`
```
Comment (souravgupta-msft) [inline component/stream/stream.go:67]: Same as previous

**Review by souravgupta-msft — APPROVED** [review]

File: cmd/mount.go (hunk 3 above)
Comment (gapra-msft) [reply cmd/mount.go:359]: The point of these is to keep them hidden/only in CLI, so adding them to base config kinda defeats the point there right?

Comment (gapra-msft) [reply cmd/mount.go:359]: Let me know if you still think we need them in baseConfig and I can add it in a follow up PR but I dont think its necessary.

**Pull request merged**